packages/aws-cdk-lib/aws-stepfunctions/lib/states/distributed-map/item-reader.ts

```diff
@@ -132,10 +165,23 @@ export class S3ObjectsItemReader implements IItemReader {
         actions: [
           's3:ListBucket',
         ],
-        resources: [this.bucket.bucketArn],
+        resources: [this._bucket ? this._bucket.bucketArn : '*'],
```

Comment:
I think my previous comment was not clear, I meant to give the permission for any bucket only if the `this.bucket` is undefined, but if it is defined, so it is better to limit the permission to that bucket.

Reply:
Yes - so above ternary operator uses bucket's ARN when bucket is given but otherwise this gives permission for any bucket. 

Am I misunderstanding something here?

Reply:
sorry, it is my bad, for some reason I did not notice it.